### ISSUE TITLE
Not reproducible after setting random seed

### ISSUE THREAD

**Issue by fanyun-sun** [rusty1s/pytorch_geometric#92]
I am running `examples/qm9_nn_conv.py`
but after setting random seed as follow, the results are still not 100% same.

```
   def seed_everything(seed=1234):                                                  
       random.seed(seed)                                                            
       torch.manual_seed(seed)                                                      
       torch.cuda.manual_seed_all(seed)                                             
       np.random.seed(seed)                                                         
       os.environ['PYTHONHASHSEED'] = str(seed)                                     
       torch.backends.cudnn.deterministic = True                                    
       torch.backends.cudnn.benchmark = False 
```

**Comment by rusty1s** [comment]
Hi and thank you for this issue.
Did you notice reproducibility issues only in `qm9_nn_conv` or everywhere?

**Comment by petwill** [comment]
I only ran `qm9_nn_conv`

**Comment by rusty1s** [comment]
I evaluated this problem and see perfect reproducibility on CPU when using the `seed_everything` code from you. However, on GPU, we can not guarantee determinism because we make heavy use of `scatter` ops to implement Graph Neural Networks [which are non-deterministic by nature on GPU](https://pytorch.org/docs/stable/notes/randomness.html).
